src/consensus/pbft_manager.cpp

```diff
@@ -1696,10 +1663,53 @@ bool PbftManager::pushPbftBlock_(SyncBlock &sync_block, vec_blk_t &dag_blocks_or
     if (calculated_order_hash != sync_block.pbft_blk->getOrderHash()) {
       LOG(log_er_) << "Order hash incorrect. Pbft order hash: " << sync_block.pbft_blk->getOrderHash()
                    << " . Calculated hash:" << calculated_order_hash;
+      std::vector<trx_hash_t> trx_order;
+      std::vector<blk_hash_t> blk_order;
+      for (auto t : sync_block.transactions) {
+        trx_order.push_back(t.getHash());
+      }
+      for (auto b : sync_block.dag_blocks) {
+        blk_order.push_back(b.getHash());
+      }
+      LOG(log_er_) << "Dag order " << blk_order;
+      LOG(log_er_) << "Trx order " << trx_order;
       return false;
     }
   }
 
+  LOG(log_nf_) << "Storing cert votes of pbft blk " << pbft_block_hash;
+  LOG(log_dg_) << "Stored following cert votes:\n" << cert_votes;
+  // Update PBFT chain head block
+  db_->addPbftHeadToBatch(pbft_chain_->getHeadHash(), pbft_chain_->getJsonStrForBlock(pbft_block_hash), batch);
+
+  if (sync) {
+    dag_blocks_order.reserve(sync_block.dag_blocks.size());
+    std::transform(sync_block.dag_blocks.begin(), sync_block.dag_blocks.end(), std::back_inserter(dag_blocks_order),
+                   [](const DagBlock &dag_block) { return dag_block.getHash(); });
+    // Update counts correctly
+
+    // Non-finalized block should be empty when syncing, maybe we should clear it if we are deep out of sync to improve
+    // performance
+    auto non_finalized_blocks = dag_mgr_->getNonFinalizedBlocks();
+    unordered_set<blk_hash_t> non_finalized_blocks_set;
```

Comment:
std::unordered_set. It is considered bad practice to have `using namespace std` and skip using `std::`. I am planning to remove this from our code so you can save me some time and write new code with std::  here and in other places